custom_components/samsungtv_tizen/media_player.py

```diff
@@ -475,36 +546,39 @@ def device_class(self):
     def turn_on(self):
         """Turn the media player on."""
         if self._power_off_in_progress():
+            self._end_of_power_off = None 
             if self._is_ws_connection:
                 self.send_command("KEY_POWER")
             else:
                 self.send_command("KEY_POWEROFF")
-            return
 
-        if self._mac:
-            if self._broadcast:
-                wakeonlan.send_magic_packet(self._mac, ip_address=self._broadcast)
+        elif self._state == STATE_OFF:
+            if self._mac:
+                if self._broadcast:
+                    wakeonlan.send_magic_packet(self._mac, ip_address=self._broadcast)
+                else:
+                    wakeonlan.send_magic_packet(self._mac)
+                time.sleep(2)
+                self._ping_device()
             else:
-                wakeonlan.send_magic_packet(self._mac)
-            time.sleep(2)
-            self._ping_device()
-        else:
-            self.send_command("KEY_POWERON")
+                self.send_command("KEY_POWERON")
 
     def turn_off(self):
         """Turn off media player."""
-        self._end_of_power_off = dt_util.utcnow() + timedelta(seconds=15)
+        if (not self._power_off_in_progress()) and self._state != STATE_OFF:
```

Comment:
Is there a reason for this check? If it's not there, then `self._power_off_in_progress()` will just reset the timer every time we turn off the TV. Why would we not want the timer to be reset every time we turn off the TV?

Reply:
If power off already in progress, we should only wait that the status reflect the action. If we enter again here in other condition, we will have the power off command to cause a power on and a not clear status on UI. Also may cause issue on automation. The var "_end_of_power_off " is finally reset during status update if really status is off.